packages/mobile/src/screens/edit-track-screen/EditTrackScreen.tsx

```diff
@@ -1,63 +1,86 @@
-import { useCallback } from 'react'
+import { useCallback, useMemo } from 'react'
 
 import type { UploadTrack } from '@audius/common'
-import { creativeCommons } from '@audius/common'
+import {
+  creativeCommons,
+  formatPrice,
+  useUSDCPurchaseConfig
+} from '@audius/common'
 import { Formik } from 'formik'
 import * as Yup from 'yup'
 
+import { useRemoteVar } from 'app/hooks/useRemoteConfig'
+
 import { EditTrackNavigator } from './EditTrackNavigator'
 import type { FormValues, EditTrackScreenProps } from './types'
 const { computeLicenseVariables, ALL_RIGHTS_RESERVED_TYPE } = creativeCommons
 
-const EditTrackSchema = Yup.object().shape({
-  title: Yup.string().required('Your track must have a name.'),
-  artwork: Yup.object({
-    url: Yup.string()
-  })
-    .when('trackArtwork', {
-      is: undefined,
-      then: Yup.object().required('Artwork is required.').nullable()
-    })
-    .nullable(),
-  trackArtwork: Yup.string().nullable(),
-  genre: Yup.string().required('Genre is required.'),
-  description: Yup.string().max(1000).nullable(),
-  premium_conditions: Yup.object({
-    usdc_purchase: Yup.object({
-      price: Yup.number()
-        .positive()
-        .min(1, 'Price must be at least $1.00.')
-        .max(9999.99, 'Price must be less than $9999.99.')
-    }).nullable()
-  }).nullable(),
-  duration: Yup.number().nullable(),
-  preview_start_seconds: Yup.number()
-    .test('isValidPreviewStart', '', function (value: number) {
-      const duration = this.resolve(Yup.ref('duration')) as unknown as number
-      // If duration is NaN, validation passes because we were
-      // unable to get duration from a track
-      if (isNaN(duration) || duration === null) return true
-      if (duration > 30 && value > duration - 30) {
-        return this.createError({
-          message:
-            'Preview must start at least 30 seconds before the end of the track.'
-        })
-      }
-      if (duration <= 30 && value !== 0) {
-        return this.createError({
-          message:
-            'Preview must start at 0 since the track is less than 30 seconds.'
+const useEditTrackSchema = () => {
```

Comment:
Would be really great to share a validation schema in the future if we can decide between Zod/Yup 😅